microsoft/testsuites/vm_extensions/run_command.py

```diff
@@ -0,0 +1,295 @@
+        settings = {
+            "source": {"CommandId": "RunShellScript", "script": f"touch {test_file}"}
+        }
+
+        extension = node.features[AzureExtension]
+        result = extension.create_or_update(
+            name="RunCommand",
+            publisher="Microsoft.CPlat.Core",
+            type_="RunCommandHandlerLinux",
+            type_handler_version="1.3",
+            auto_upgrade_minor_version=True,
+            settings=settings,
+        )
+
+        assert_that(result["provisioning_state"]).described_as(
+            "Expected the extension to succeed"
+        ).is_equal_to("Succeeded")
+
+        message = f"File {test_file} was not created on the test machine"
+        # Verify that file was created on the test machine
+        node.execute(
+            f"ls '{test_file}'",
+            shell=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=message,
+        )
+
+    @TestCaseMetadata(
+        description="""
+        Runs the Run Command v2 VM extension with a public uri pointing to the
+        script in blob storage.
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
+    )
+    def verify_public_uri_script_run(self, log: Logger, node: Node) -> None:
+        script_uri = "https://rcv2lisa.blob.core.windows.net/publiccontainer/test.sh"
+        test_file = "/tmp/lisatest.txt"
+        settings = {
+            "source": {
+                "CommandId": "RunShellScript",
+                "scriptUri": script_uri,
+            },
+        }
+
+        extension = node.features[AzureExtension]
+        result = extension.create_or_update(
+            name="RunCommand",
+            publisher="Microsoft.CPlat.Core",
+            type_="RunCommandHandlerLinux",
+            type_handler_version="1.3",
+            auto_upgrade_minor_version=True,
+            settings=settings,
+        )
+
+        assert_that(result["provisioning_state"]).described_as(
+            "Expected the extension to succeed"
+        ).is_equal_to("Succeeded")
+
+        message = f"File {test_file} was not created on the test machine"
+        # Verify that file was created on the test machine
+        node.execute(
+            f"ls '{test_file}'",
+            shell=True,
+            expected_exit_code=0,
+            expected_exit_code_failure_message=message,
+        )
+
+    @TestCaseMetadata(
+        description="""
+        Runs the Run Command v2 VM extension with a private storage uri pointing
+        to the script in blob storage. No sas token provided, should fail.
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_features=[AzureExtension]),
+    )
+    def verify_private_uri_script_run_failed(
+        self, log: Logger, node: Node, environment: Environment
+    ) -> None:
+        platform = environment.platform
+        assert isinstance(platform, AzurePlatform)
```

Comment:
It can be added as a platform requirement in requirement metadata. So the test case will be skipped on other platforms directly.